<?xml version="1.0"?>
<launch>
  
  <node 
    pkg="turtlesim" 
    type="turtlesim_node"
    name="turtlesim" 
    respawn="true"
  /> 

  <!-- Starting robot state publish which will publish tf -->
  <node 
    name="turtle_teleop" 
    pkg="turtlesim" 
    type="turtle_teleop_key"
    launch-prefix="xterm -e"
    required="true"
  />

  <node 
    name="distace_checker" 
    pkg="beginner_tutorials" 
    type="distance_checker_plus" 
  />
  <remap from="turtle1/cmd_vel" to="turtle2/cmd_vel"/>
  <node pkg="turtlesim" name="turtle2" type="turtle_teleop_key" output="screen" launch-prefix="xterm -e"/>
    

  <node 
    name="distace_observer" 
    pkg="beginner_tutorials" 
    type="distance_observer"
    output="screen" 
    launch-prefix="xterm -e"
  />
</launch>
